### PR TITLE
Fix Coveralls submission, and re-enable Coveralls submission from CI

### LINKED ISSUE #314: Debugging the Coveralls errors

### PR DESCRIPTION
**Pull request by DilumAluthge**
Fixes #314 
Closes #310 
Closes #313

### CODE REVIEW

File: src/coveralls.jl [hunk 1]
```diff
@@ -102,7 +110,9 @@ module Coveralls
             event_path = open(JSON.Parser.parse, ENV["GITHUB_EVENT_PATH"])
             github_pr_info = get(event_path, "pull_request", Dict())
             github_pr = get(github_pr_info, "number", "")
-            isempty(github_pr) || (data["service_pull_request"] = github_pr)
+            github_pr_string = String(strip(_stringify_string_or_int(github_pr)))
+            github_pr_string::String
+            isempty(github_pr) || (data["service_pull_request"] = github_pr_string)
```
Comment (vtjnash) [inline src/coveralls.jl]:
```suggestion
            isempty(github_pr) || (data["service_pull_request"] = strip(string(github_pr)))
```

File: src/coveralls.jl [hunk 2]
```diff
@@ -92,7 +100,7 @@ module Coveralls
 
             # get the name of the branch if not a pull request
             if get(ENV, "CI_PULL_REQUEST", "false") == "false"
-                data["git"]["branch"] = split(ENV["GIT_BRANCH"], "/")[2]
+                data["git"]["branch"] = String(split(ENV["GIT_BRANCH"], "/")[2])
```
Comment (vtjnash) [inline src/coveralls.jl:103]: I expect the JSON formatting should already handle AbstractString, SubString, and String the same

File: src/coveralls.jl (hunk 1 above)
Comment (DilumAluthge) [reply src/coveralls.jl]: Oops I just pushed something that made this review "outdated", so I can't click the "commit suggestion" button, but I'll make that change manually and commit it.

**Review by vtjnash — APPROVED** [review]
Thanks for tracking that down!

File: src/coveralls.jl [hunk 3]
```diff
@@ -66,6 +66,14 @@ module Coveralls
...
     end
 
+    function _stringify_string_or_int(str::AbstractString)
+        return str
+    end
+
+    function _stringify_string_or_int(int::Integer)
+        return string(int)
+    end
+
```
Comment (vtjnash) [inline src/coveralls.jl:76]:
```suggestion
```

File: src/coveralls.jl (hunk 2 above)
Comment (DilumAluthge) [reply src/coveralls.jl:103]: Earlier I was running into issues when it was a `SubString`, but I was debugging, so I had changed a bunch of stuff. Let me get rid of the `String()` here and let's see if it still works.

**Comment by DilumAluthge** [comment]
> Thanks for tracking that down!

Your tip to look at https://github.com/lemurheavy/coveralls-public/issues/1527 was spot-on. The issue was indeed that our JSON was not typed the way that Coveralls expected it to be typed.

File: src/coveralls.jl (hunk 2 above)
Comment (DilumAluthge) [reply src/coveralls.jl]: Done.

**Pull request merged**